native_extension/GoogleSignInANE/src/com/tuarua/google/signin/GoogleSignInAccount.as

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2021 Tua Rua Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.tuarua.google.signin {
+
+/**
+ * GoogleSignIn account information
+ */
+public class GoogleSignInAccount {
+
+    private var _id:String;
+    private var _idToken:String;
+    private var _serverAuthCode:String;
+    private var _displayName:String;
+    private var _email:String;
+    private var _photoUrl:String;
+    private var _grantedScopes:Array;
+    private var _familyName:String;
+    private var _givenName:String;
+    private var _accessToken:String;
+
+    /**
+     * @param {Object} data
+     * @param {String} [data.id = null]
+     * @param {String} [data.idToken = null]
+     * @param {String} [data.serverAuthCode = null]
+     * @param {String} [data.displayName = null]
+     * @param {String} [data.email = null]
+     * @param {String} [data.photoUrl = null]
+     * @param {String[]} [data.grantedScopes = null]
+     * @param {String} [data.familyName = null]
+     * @param {String} [data.givenName = null]
+     * @param {String} [data.accessToken = null]
+     */
+    public function GoogleSignInAccount(data:Object) {
```

Comment:
I don't like passing objects into constructors.
Getters and setters should be used with default property values set when the private var is declared.